extends Area2D

@export var door_color: int = 7 # m√†u c·ª≠a
@onready var player: CharacterBody2D = get_tree().get_first_node_in_group("Player")
@onready var tilemap = $"../TileMap"

var is_open: bool = false

@export var tiles_to_remove: Array[Vector2i] = [
	Vector2i(19, -9),
	Vector2i(19, -10),
	Vector2i(19, -11),
	Vector2i(19, -12)
]

func _on_body_entered(body):
	if body.is_in_group("Player") and not is_open:
		if body.current_color == door_color:
			print("C·ª≠a m·ªü th√†nh c√¥ng üéâ")
			open_door()
			body.reset_color()
		else:
			print("Sai m√†u ‚Üí player ch·∫øt üíÄ")
			body.die()

func open_door():
	is_open = true
	# X√≥a tile c·ª≠a
	for coords in tiles_to_remove:
		tilemap.set_cell(0, coords, -1)

	# Sau 1 gi√¢y c·ª≠a ƒë√≥ng l·∫°i
	await get_tree().create_timer(1.0).timeout
	close_door()

func close_door():
	is_open = false
	# ƒê·∫∑t l·∫°i tile c·ª≠a b·∫±ng ID ƒë√£ hardcode s·∫µn
	tilemap.set_cell(0, Vector2i(19, -9), 0, Vector2i(20, 8))
	for coords in tiles_to_remove:
		tilemap.set_cell(0, coords, 0, Vector2i(20, 9))
	tilemap.set_cell(0, Vector2i(19, -12), 0, Vector2i(20, 10))
